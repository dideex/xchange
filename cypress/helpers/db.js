const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
