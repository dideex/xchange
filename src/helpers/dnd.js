import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
