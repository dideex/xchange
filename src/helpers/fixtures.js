export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}