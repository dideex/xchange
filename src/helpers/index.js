export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
