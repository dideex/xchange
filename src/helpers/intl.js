import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
