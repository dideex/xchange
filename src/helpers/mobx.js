import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
