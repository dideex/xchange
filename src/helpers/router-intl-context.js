import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
