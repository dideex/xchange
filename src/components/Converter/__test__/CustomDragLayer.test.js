import React from 'react'
import {mount, shallow} from 'enzyme'
import {DragDropContext, DragSource} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'
import PropTypes from 'prop-types'

import Component from '../CustomDragLayer'
import {wrapInTestContext} from '../../../helpers/dnd'

describe('Coverter: Currency badge tests', () => {
  it("Shouldn't render if wasn't dragging", () => {
    const WrappedComponent = wrapInTestContext(Component)

    const wrapper = mount(<WrappedComponent />)
    expect(wrapper.html()).toMatchSnapshot()
  })
  describe('Preview behavriour', () => {
    const createComponent = ({generator = jest.fn(), source = null} = {}) => {
      @DragDropContext(TestBackend)
      class TestRoot extends React.Component {
        render() {
          return (
            <div>
              {source}
              <Component generator={generator} />
            </div>
          )
        }
      }

      return mount(<TestRoot />)
    }

    it('Basic markup', () => {
      const Source =
        @DragSource(
          'cash',
          {
            beginDrag: () => {
              return {coucou: 'cash'}
            },
            canDrag: () => {
              return true
            },
          },
          connect => {
            return {connectDragSource: connect.dragSource()}
          },
        )
        class DS extends React.Component {
          static propTypes = {connectDragSource: PropTypes.func}
          render() {
            return this.props.connectDragSource(<div />)
          }
        }

      const wrapper = createComponent({source: <Source />})

      const backend = wrapper
        .instance()
        .getManager()
        .getBackend()
      backend.simulateBeginDrag(
        [
          wrapper
            .find(Source)
            .instance()
            .getHandlerId(),
        ],
        {
          clientOffset: {x: 1, y: 2},
          getSourceClientOffset: () => {
            return {x: 1000, y: 2000}
          },
        },
      )

      wrapper.update()

      // const backend = wrapper
      //   .instance()
      //   .getManager()
      //   .getBackend()
      // const ctx = wrapper.instance().getChildContext()
      // console.log('TCL: backend', ctx)

      // wrapper.find('CustomDragLayer').render()
      console.log(wrapper.debug())
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})

/* 
  
      const backend = wrapper
        .instance()
        .getManager()
        .getBackend()

  // Test that the opacity is 1
  let div = TestUtils.findRenderedDOMComponentWithTag(root, 'div')
  expect(div.props.style.opacity).toEqual(1)

  // Find the drag source ID and use it to simulate the dragging operation
  const box = TestUtils.findRenderedComponentWithType(root, Box)
  backend.simulateBeginDrag([box.getHandlerId()])

  // Verify that the div changed its opacity
  div = TestUtils.findRenderedDOMComponentWithTag(root, 'div')
  expect(div.style.opacity).toEqual(0.4)
  */
