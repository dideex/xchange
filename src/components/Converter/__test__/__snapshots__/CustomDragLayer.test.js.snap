// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coverter: Currency badge tests Preview behavriour Basic markup 1`] = `"<div><div></div></div>"`;

exports[`Coverter: Currency badge tests Shouldn't render if wasn't dragging 1`] = `null`;
